dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -458,8 +458,7 @@ public Object visitNreC11Store(LitmusCParser.NreC11StoreContext ctx) {
     public Object visitNreAssignment(LitmusCParser.NreAssignmentContext ctx){
         Expression variable = (Expression)ctx.varName().accept(this);
         if(ctx.Ast() == null){
-            if(variable instanceof Register){
-                returnRegister = (Register)variable;
+            if(variable instanceof Register returnRegister){
                 ctx.re().accept(this);
                 return null;
```

Comment:
This is wrong, because the original code wrote into an instance variable, while the new code creates a fresh and unused local.